ci/jenkins/utils.sh

```diff
@@ -14,6 +14,23 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+function check_and_cleanup_buildkit_cache() {
+    free_space=$(df -h -B 1G / | awk 'NR==2 {print $4}')
+    free_space_threshold=40
+    if [[ $free_space -lt $free_space_threshold ]]; then
+        # If cleaning up unused dangling images doesn't free up sufficient disk space,
+        # we will have to reduce the buildkit cache to 10GB to release enough disk space.
+        docker buildx prune -af --keep-storage=10gb > /dev/null
+        free_space=$(df -h -B 1G / | awk 'NR==2 {print $4}')
+        if [[ $free_space -lt $free_space_threshold ]]; then
+            # If the first round cleanup doesn't free up sufficient disk space,
+            # we will have to clean up all buildkit cache to release enough disk space.
+            docker buildx prune -af > /dev/null
+        fi
+    fi
+    docker buildx du
+}
```

Comment:
I thought we considered some logic like this previously but decided not to use it? Am I misremembering?

Reply:
For the previous code changes, we don't want to remove all images because doing so may delete necessary images for other concurrent jobs on the same testbed. However, for the BuildKit cache, it can be safely deleted without encountering any build image errors. The only difference is the building speed.

Reply:
So this is because there is no equivalent command to `docker buildx prune` in legacy docker? Is `docker builder prune` different from this one?

I wonder if we should just unconditionally run `docker buildx prune -af --keep-storage=10gb` instead of checking for free space first?

Reply:
> So this is because there is no equivalent command to `docker buildx prune` in legacy docker? Is `docker builder prune` different from this one?
I tried both commands and docker builder prune can achieve the same result of docker buildx prune. 

> I wonder if we should just unconditionally run `docker buildx prune -af --keep-storage=10gb` instead of checking for free space first?

Updated, I am okay with both methods.

Reply:
I meant only using `docker builder prune -af --keep-storage=10gb`, and removing the rest of the logic. However, I am fine with the current approach. Let's see if it eliminates our space issues.